src/pybind/mgr/orchestrator.py

```diff
@@ -384,36 +386,158 @@ class DriveGroupSpec(object):
     """
     Describe a drive group in the same form that ceph-volume
     understands.
+    This structure allows the selection of storage devices
+    in a set of nodes
     """
-    def __init__(self, devices):
-        self.devices = devices
+    def _init_attribute(self, att_dict, attribute_name, default_value):
+        """ Extract attribute from dict, if not present applies default value
+
+        :param att_dict      : dictionary with attributes
+        :param attribute_name: The attribute to initialize
+        :param default_value : Value applied if the attribute is not in the dict
+        """
+        if attribute_name in att_dict.keys():
+            value = att_dict[attribute_name]
+        else:
+            value = default_value
+
+        setattr(self, attribute_name, value)
+
+    def _verify_device_especification(self, dev_spec):
+        """Verifies a drive group device specification
+
+        Attributes supported in the device specification are:
+            {
+            id_model : A wildcard string. e.g: "SDD*"
+            size     : Size specification of format LOW:HIGH.
+                        Can also take the the form :HIGH, LOW:
+                        or an exact value (as ceph-volume inventory reports)
+            rotates  : is the drive rotating or not
+            count    : if this is present limit the number of drives to this
+                        number.
+            }
+
+        Any attributes (even none) can be included in the device
+        specification structure.
+
+        :param device_spec: Dict with the device specification desired
+        """
+        # A dict needed
+        if type(dev_spec) is not dict:
```

Comment:
Actually a `dict` is not strictly required. I'd assume it's something dict-like. I don't see a strong reason to type check this here.

Reply:
When we will know what is the exact type of dict we need, we can replace this check. I think that is not bad to do this kind of check ... to pass a json/yaml string instead a dict-like could be easy ...